cmd/influx/main.go

```diff
@@ -91,8 +92,31 @@ func newApp() cli.App {
 	}
 }
 
+// This replaces `-t "-FOO-TOKEN"` with `-t=-FOO-TOKEN` in os.Args
+// This is necessary to do because the command line arg:
+//  `-t "-FOO-TOKEN"`` will be parsed as two separate flags instead of a flag and token value.
+func replaceTokenArg() {
+	for i, arg := range os.Args {
+		switch arg {
+		case "--token", "-t":
+			// if last element, this will be invalid later
+			if i == len(os.Args)-1 {
+				break
+			}
+			os.Args[i] = strings.Join(os.Args[i:i+2], "=")
+			// if there are 2+ elements after this
+			if len(os.Args) > i+2 {
+				os.Args = append(os.Args[:i+1], os.Args[i+2:]...)
+			} else {
+				os.Args = os.Args[:i+1]
+			}
+		}
+	}
+}
+
 func main() {
 	app := newApp()
+	replaceTokenArg()
```

Comment:
This is clever! Can we create a new slice instead of messing with the os.Args one though (and pass that to `app.Run`?)